### PR TITLE
Add `sql_hook_params` parameter to `S3ToSqlOperator`

### PR DESCRIPTION
**Pull request by alexbegg**
Adding `sql_hook_params` parameter to `SqlToS3Operator`. This will allow you to pass extra config params to the underlying SQL hook.

This uses the same "sql_hook_params" parameter name as already used in `SqlToSlackOperator`.

(This is related to https://github.com/apache/airflow/pull/33425 which adds `sql_hook_params` to the opposite transfer, `SqlToS3Operator`)

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by alexbegg** [comment]
@uranusjr This PR retains how `S3ToSqlOperator ` accesses `BaseHook`, but adds a way to pass `hook_params` parameter to `BaseHook.get_hook`, which then calls `Connection.get_hook` (which itself already accepts `hook_params`).

File: airflow/providers/amazon/aws/transfers/s3_to_sql.py [hunk 1]
```diff
@@ -120,7 +124,7 @@ def execute(self, context: Context) -> None:
     @cached_property
     def db_hook(self):
         self.log.debug("Get connection for %s", self.sql_conn_id)
-        hook = BaseHook.get_hook(self.sql_conn_id)
+        hook = BaseHook.get_hook(self.sql_conn_id, hook_params=self.sql_hook_params)
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/transfers/s3_to_sql.py]: You can’t do this because providers needs to support older Airflow versions.

Comment (alexbegg) [reply airflow/providers/amazon/aws/transfers/s3_to_sql.py]: You are right, I am going to switch this to a draft and look into a different approach

**Comment by alexbegg** [comment]
@uranusjr I corrected the test to access the hook using the changed method of getting it from `Connection.get_hook` (which supports `hook_params`) instead of `BaseHook.get_hook` (which doesn't), the test passes now.

**Comment by alexbegg** [comment]
Please remove [area:core-operators](https://github.com/apache/airflow/labels/area%3Acore-operators) label from this PR now no longer changes the core operators

**Review by potiuk — APPROVED** [review]

**Pull request merged**